django/test/testcases.py

```diff
@@ -898,12 +902,19 @@ def assertNumQueries(self, num, func=None, *args, **kwargs):
             func(*args, **kwargs)
 
 
-def connections_support_transactions():
+def is_dummy_db_wrapper(connections, alias):
+    return isinstance(connections[alias].features.connection, DummyDatabaseWrapper)
+
+
+def connections_support_transactions(_connections=None):
     """
     Returns True if all connections support transactions.
     """
+    _connections = _connections or connections
+    if any(is_dummy_db_wrapper(_connections, conn.alias) for conn in _connections.all()):
```

Comment:
It might be better to set `supports_transactions = False` on the dummy backend's `DatabaseFeatures` class so we don't need special handing for it here.